ledger/participant-integration-api/src/main/scala/platform/store/backend/common/QueryStrategy.scala

```diff
@@ -55,4 +55,6 @@ trait QueryStrategy {
 
   /** Boolean predicate */
   def isTrue(booleanColumnName: String): String
+
+  def nullAs(typeName: String): String = s"NULL::$typeName"
```

Comment:
Two comments:
- the double colon operator is not standard, I would not provide a default implementation
- type names are not portable - see for example `BIGINT` vs `NUMBER`

How about adding a `def nullAsNumber: string` instead? Then you wouldn't need the overrideable `NullAsLedgerEffectiveTimeType`.